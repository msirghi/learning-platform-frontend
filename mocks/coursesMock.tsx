import { Course } from '../common/types';

export const coursesMock: Course[] = [
  {
    id: '1',
    name: 'Course 1',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Libero non illum est fuga beatae ab, assumenda reiciendis nesciunt numquam similique!`,
    coverImg:
      'https://upload.wikimedia.org/wikipedia/commons/thumb/b/b6/Image_created_with_a_mobile_phone.png/1200px-Image_created_with_a_mobile_phone.png',
    teacherImg:
      'https://dyl80ryjxr1ke.cloudfront.net/external_assets/hero_examples/hair_beach_v1785392215/original.jpeg'
  },
  {
    id: '2',
    name: 'Course 2',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    coverImg: 'https://www.w3schools.com/w3css/img_lights.jpg',
    teacherImg:
      'https://images.pexels.com/photos/614810/pexels-photo-614810.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500'
  },
  {
    id: '3',
    name: 'Course 3',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    coverImg:
      'https://helpx.adobe.com/content/dam/help/en/stock/how-to/visual-reverse-image-search/jcr_content/main-pars/image/visual-reverse-image-search-v2_intro.jpg',
    teacherImg:
      'https://i2-prod.mirror.co.uk/incoming/article14334083.ece/ALTERNATES/s1200c/3_Beautiful-girl-with-a-gentle-smile.jpg'
  },
  {
    id: '4',
    name: 'Course 4',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    coverImg:
      'https://www.metoffice.gov.uk/binaries/content/gallery/metofficegovuk/hero-images/advice/maps-satellite-images/satellite-image-of-globe.jpg',
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '5',
    name: 'Course 5',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '6',
    name: 'Course 6',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '7',
    name: 'Course 7',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '8',
    name: 'Course 8',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '9',
    name: 'Course 9',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '10',
    name: 'Course 10',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '11',
    name: 'Course 11',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '12',
    name: 'Course 12',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '13',
    name: 'Course 13',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  }
];
