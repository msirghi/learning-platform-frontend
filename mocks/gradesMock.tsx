import { GradeType } from '../common/enums';
import { Grade } from '../common/types';

export const GradesMock: Grade[] = [
  {
    id: '1',
    course: 'Mathematics',
    marks: [
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_1,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_2,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.EXAM,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.INDIVIDUAL_WORK,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.INDIVIDUAL_WORK,
        updatedAt: new Date()
      }
    ],
    teacher: 'Teacher 1',
    year: 'First year',
    maxPresence: 15,
    presence: 3
  },
  {
    id: '2',
    course: 'Computer Science Computer Science',
    marks: [
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_1,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_2,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.EXAM,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.INDIVIDUAL_WORK,
        updatedAt: new Date()
      }
    ],
    teacher: 'Teacher 2',
    year: 'First year',
    maxPresence: 15,
    presence: 7
  },
  {
    id: '3',
    course: 'Biology',
    marks: [
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_1,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_2,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.EXAM,
        updatedAt: new Date()
      }
    ],
    teacher: 'Teacher 3',
    year: 'First year',
    maxPresence: 20,
    presence: 15
  },
  {
    id: '4',
    course: 'Physics',
    marks: [
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_1,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_2,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.EXAM,
        updatedAt: new Date()
      }
    ],
    teacher: 'Teacher 4',
    year: 'First year',
    maxPresence: 20,
    presence: 19
  }
];
