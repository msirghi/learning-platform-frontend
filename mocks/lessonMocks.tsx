import { LessonStatus } from '../common/enums';
import { Lesson } from '../common/types';

export const lessonMock: Lesson[] = [
  {
    id: '22',
    status: LessonStatus.ENDED,
    topic: 'Topic 1',
    title: `Lorem ipsum dolor sit amet.`,
    time: '8:00',
    start: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      10,
      0,
      0,
      0
    ),
    end: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      10,
      30,
      0,
      0
    )
  },
  {
    id: '2',
    status: LessonStatus.ONGOING,
    topic: `Lorem ipsum dolor sit amet.`,
    title: 'Lesson 2',
    time: '9:30',
    start: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      13,
      0,
      0,
      0
    ),
    end: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      14,
      0,
      0,
      0
    )
  },
  {
    id: '3',
    status: LessonStatus.NOT_STARTED,
    topic: 'Topic 3',
    title: 'Lesson 3',
    time: '11:00',
    start: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      14,
      30,
      0,
      0
    ),
    end: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      15,
      30,
      0,
      0
    )
  },
  {
    id: '4',
    status: LessonStatus.NOT_STARTED,
    topic: 'Topic 4',
    title: 'Lesson 4',
    time: '13:00',
    start: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      16,
      0,
      0,
      0
    ),
    end: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      17,
      30,
      0,
      0
    )
  }
];
