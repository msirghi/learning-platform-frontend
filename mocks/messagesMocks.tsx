import { ChatMessage } from '../common/types';

export const messagesMock: ChatMessage[] = [
  {
    id: 1,
    author: 'Mihail',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet.`
  },
  {
    id: 2,
    author: 'Nicolae',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Praesentium, sapiente!`
  },
  {
    id: 3,
    author: 'Mihail 2',
    date: new Date(),
    message: `Lorem ipsum, dolor sit amet consectetur adipisicing.`
  },
  {
    id: 4,
    author: 'Mihail 4',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus, repudiandae? Iusto labore dolores obcaecati maxime. Quaerat praesentium doloribus expedita voluptatum!`
  },
  {
    id: 5,
    author: 'Mihail 4',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus, repudiandae? Iusto labore dolores obcaecati maxime. Quaerat praesentium doloribus expedita voluptatum!`
  },
  {
    id: 6,
    author: 'Mihail 4',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus, repudiandae? Iusto labore dolores obcaecati maxime. Quaerat praesentium doloribus expedita voluptatum!`
  }
];
