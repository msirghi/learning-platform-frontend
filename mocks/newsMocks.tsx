import { News } from "../common/types";

export const newsMock: News[] = [
  {
    id: "1",
    date: new Date(),
    description: `Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dignissimos nihil, commodi alias at debitis in.`,
    title: "News 1",
  },
  {
    id: "2",
    date: new Date(),
    description: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Officiis numquam nobis, nihil quis error repudiandae.`,
    title: "News 2",
  },
];
