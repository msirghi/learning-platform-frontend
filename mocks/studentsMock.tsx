import { Student } from '../common/types';

export const studentsMock: Student[] = [
  {
    id: 1,
    firstName: 'First',
    lastName: 'Last'
  },
  {
    id: 2,
    firstName: 'Nicolae',
    lastName: 'Budulae'
  },
  {
    id: 3,
    firstName: 'Budulae',
    lastName: 'Nicolae'
  },
  {
    id: 4,
    firstName: 'Aron',
    lastName: 'Pumnul'
  },
  {
    id: 5,
    firstName: 'First',
    lastName: 'Last'
  },
  {
    id: 6,
    firstName: 'First',
    lastName: 'Last'
  },
  {
    id: 7,
    firstName: 'First',
    lastName: 'Last'
  }
];
