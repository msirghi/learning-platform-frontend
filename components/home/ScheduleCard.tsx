import React from 'react';
import styles from '../../styles/modules/Home.module.scss';
import { LessonRow } from './LessonRow';
import { LessonStatus } from '../../common/enums';
import { lessonMock } from '../../mocks/lessonMocks';
import { useTranslation } from '../../i18n';

export const ScheduleCard: React.FC = () => {
  const { t } = useTranslation();

  return (
    <div className={styles.scheduleContainer}>
      <div className={styles.scheduleHeader}>
        <div className={styles.scheduleDate}>
          <div className={styles.scheduleDateNumber}>{new Date().getDate()}</div>
          <div className={styles.scheduleDateMonth}>
            {new Date().toLocaleDateString('default', { month: 'short' })}
          </div>
        </div>
        <div className={styles.scheduleHeaderText}>
          <div>{t('home:scheduleCard.welcomeBack')}, Nicolay.</div>
          <div>
            {t('home:scheduleCard.todayYouHave')} 4 {t('home:scheduleCard.lessons')}.
          </div>
        </div>
      </div>

      <div className={styles.scheduleContent}>
        <div className={styles.scheduleContentHeader}>{t('home:scheduleCard.scheduleForThisDay')}:</div>
        {lessonMock.map((lesson, idx) => {
          let lineClassname = styles.scheduleContentVerticalLine;
          if (lesson.status !== LessonStatus.ENDED) {
            lineClassname += ` ${styles.scheduleContentVerticalLineNotEnded}`;
          }
          return (
            <div key={lesson.id} className={styles.scheduleContentLessonRow}>
              <LessonRow lesson={lesson} />
              {idx !== lessonMock.length - 1 && <div className={lineClassname} />}
            </div>
          );
        })}
      </div>
    </div>
  );
};
