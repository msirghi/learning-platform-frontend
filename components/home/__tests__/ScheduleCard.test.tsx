import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { lessonMock } from '../../../mocks/lessonMocks';
import { ScheduleCard } from '../ScheduleCard';

describe('ScheduleCard component', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<ScheduleCard />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should mock.length - 1 vertical lines', () => {
    const wrapper = shallow(<ScheduleCard />);
    const lines = wrapper.find('.scheduleContentVerticalLine');
    expect(lines.length).toBe(lessonMock.length - 1);
  });

  it('should have vertical line with specific classname if lesson has not ended', () => {
    const wrapper = shallow(<ScheduleCard />);
    const lines = wrapper.find('.scheduleContentVerticalLineNotEnded');
    expect(lines.length).toBe(2);
  });
});
