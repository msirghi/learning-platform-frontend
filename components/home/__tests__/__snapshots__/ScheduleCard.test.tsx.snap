// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleCard component should match the snapshot 1`] = `
<div
  className="scheduleContainer"
>
  <div
    className="scheduleHeader"
  >
    <div
      className="scheduleDate"
    >
      <div
        className="scheduleDateNumber"
      >
        22
      </div>
      <div
        className="scheduleDateMonth"
      >
        Nov
      </div>
    </div>
    <div
      className="scheduleHeaderText"
    >
      <div>
        home:scheduleCard.welcomeBack
        , Nicolay.
      </div>
      <div>
        home:scheduleCard.todayYouHave
         4 
        home:scheduleCard.lessons
        .
      </div>
    </div>
  </div>
  <div
    className="scheduleContent"
  >
    <div
      className="scheduleContentHeader"
    >
      home:scheduleCard.scheduleForThisDay
      :
    </div>
    <div
      className="scheduleContentLessonRow"
      key="22"
    >
      <LessonRow
        lesson={
          Object {
            "end": 2000-11-22T08:30:00.000Z,
            "id": "22",
            "start": 2000-11-22T08:00:00.000Z,
            "status": "ended",
            "time": "8:00",
            "title": "Lorem ipsum dolor sit amet.",
            "topic": "Topic 1",
          }
        }
      />
      <div
        className="scheduleContentVerticalLine"
      />
    </div>
    <div
      className="scheduleContentLessonRow"
      key="2"
    >
      <LessonRow
        lesson={
          Object {
            "end": 2000-11-22T12:00:00.000Z,
            "id": "2",
            "start": 2000-11-22T11:00:00.000Z,
            "status": "ongoing",
            "time": "9:30",
            "title": "Lesson 2",
            "topic": "Lorem ipsum dolor sit amet.",
          }
        }
      />
      <div
        className="scheduleContentVerticalLine scheduleContentVerticalLineNotEnded"
      />
    </div>
    <div
      className="scheduleContentLessonRow"
      key="3"
    >
      <LessonRow
        lesson={
          Object {
            "end": 2000-11-22T13:30:00.000Z,
            "id": "3",
            "start": 2000-11-22T12:30:00.000Z,
            "status": "notStarted",
            "time": "11:00",
            "title": "Lesson 3",
            "topic": "Topic 3",
          }
        }
      />
      <div
        className="scheduleContentVerticalLine scheduleContentVerticalLineNotEnded"
      />
    </div>
    <div
      className="scheduleContentLessonRow"
      key="4"
    >
      <LessonRow
        lesson={
          Object {
            "end": 2000-11-22T15:30:00.000Z,
            "id": "4",
            "start": 2000-11-22T14:00:00.000Z,
            "status": "notStarted",
            "time": "13:00",
            "title": "Lesson 4",
            "topic": "Topic 4",
          }
        }
      />
    </div>
  </div>
</div>
`;
